### PR TITLE
Fixup middleware codegen with the correct pointer value types

### PR DESCRIPTION
**Pull request by jasdel**
Fixes up the middleware codegen so that it uses the correct pointer value types where needed when adding middleware.

Adds a HTTPSigner to the client's options for the request signer.

Related to https://github.com/awslabs/smithy-go/pull/78

### CODE REVIEW

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java [hunk 1]
```diff
@@ -68,6 +67,13 @@
                             + "recoverable failures. When nil the API client will use a default\n"
                             + "retryer.")
                     .build(),
+            ConfigField.builder()
+                    .name(HTTP_SIGNER_CONFIG_NAME)
+                    .type(getAwsSignerV4Symbol("HTTPSigner"))
+                    .documentation("Retryer guides how HTTP requests should be retried in case of\n"
+                            + "recoverable failures. When nil the API client will use a default\n"
+                            + "retryer.")
```
Comment (skmcgrail) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java:75]: Docstring is incorrect.

Comment (jasdel) [reply codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java]: Thanks fixed.

File: go.sum [hunk 2]
```diff
@@ -2,12 +2,15 @@ github.com/awslabs/smithy-go v0.0.0-20200421200441-f1e89484c1b9 h1:oNbA/uNHusPiG
...
 github.com/awslabs/smithy-go v0.0.0-20200604194311-25e885347bc8/go.mod h1:L4SfPH3TPbKwyBENwHDh61AAQPvFh5wR00tNeUR7OrU=
+github.com/awslabs/smithy-go v0.0.0-20200617230012-dcb49a2a302d h1:kLefSezMjHYpy9ayv5M7trxeERY8HV94m4n+b4tXqi0=
+github.com/awslabs/smithy-go v0.0.0-20200617230012-dcb49a2a302d/go.mod h1:hPOQwnmBLHsUphH13tVSjQhTAFma0/0XoZGbBcOuABI=
 github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/go-sql-driver/mysql v1.5.0 h1:ozyZYNQW3x3HtqT1jira07DN2PArx2v7/mN66gGcHOs=
 github.com/go-sql-driver/mysql v1.5.0/go.mod h1:DCzpHaOWr8IXmIStZouvnhqoel9Qv2LBy8hT2VhHyBg=
 github.com/google/go-cmp v0.4.0 h1:xsAVV57WRhGj6kEIi8ReJzQlHHqcBYCElAvkovg3B/4=
 github.com/google/go-cmp v0.4.0/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
+github.com/google/go-cmp v0.4.1/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
```
[the remark this answers is not in the document]
Comment (skotambkar) [reply go.sum:13]: https://github.com/aws/aws-sdk-go-v2/pull/596/files#diff-f949e2d81c8076ebbf8af38fcbb72c1fR7-R8

[the remark this answers is not in the document]
Comment (jasdel) [reply go.sum:8]: Not sure how go mod handles the sum files with this regard. I ran `go mod tidy` and it looked like it cleaned up the go.sum file.

**Pull request merged**